frontend/src/modules/WellLogViewer/settings/components/TemplateTrackSettings/SortablePlotList.tsx

```diff
@@ -0,0 +1,228 @@
+            });
+
+            onUpdatePlots([...props.plots, plotConfig]);
+        },
+        [onUpdatePlots, props.plots]
+    );
+
+    const removePlot = React.useCallback(
+        function removePlot(plot: TemplatePlotConfig) {
+            onUpdatePlots(props.plots.filter((p) => p._id !== plot._id));
+        },
+        [onUpdatePlots, props.plots]
+    );
+
+    const handlePlotUpdate = React.useCallback(
+        function handlePlotUpdate(newPlot: TemplatePlotConfig) {
+            const newPlots = props.plots.map((p) => (p._id === newPlot._id ? newPlot : p));
+
+            onUpdatePlots(newPlots);
+        },
+        [onUpdatePlots, props.plots]
+    );
+
+    const handleTrackMove = React.useCallback(
+        function handleTrackMove(
+            movedItemId: string,
+            originId: string | null,
+            destinationId: string | null,
+            newPosition: number
+        ) {
+            // Skip update if the item was moved above or below itself, as this means no actual move happened
+            // TODO: This should probably be checked inside SortableList
+            const currentPosition = props.plots.findIndex((p) => p.name === movedItemId);
+            if (currentPosition === newPosition || currentPosition + 1 === newPosition) return;
+
+            const newTrackCfg = arrayMove(props.plots, currentPosition, newPosition);
+
+            onUpdatePlots(newTrackCfg);
+        },
+        [onUpdatePlots, props.plots]
+    );
+
+    return (
+        <div className="">
+            <Label text="Plots" position="left" wrapperClassName="!justify-between" labelClassName="!mb-0">
+                <AddItemButton buttonText="Add plot" options={PLOT_TYPE_OPTIONS} onOptionClicked={addPlot} />
+            </Label>
+
+            <SortableList onItemMoved={handleTrackMove}>
+                {props.plots.map((plot) => (
+                    <SortablePlotItem
+                        key={plot._id}
+                        plot={plot}
+                        curveHeaderOptions={curveHeaderOptions}
+                        onPlotUpdate={handlePlotUpdate}
+                        onDeletePlot={removePlot}
+                    />
+                ))}
+            </SortableList>
+        </div>
+    );
+}
+
+type SortablePlotItemProps = {
+    plot: TemplatePlotConfig;
+    curveHeaderOptions: DropdownOption[];
+    onPlotUpdate: (plot: TemplatePlotConfig) => void;
+    onDeletePlot: (plot: TemplatePlotConfig) => void;
+};
+
+function SortablePlotItem(props: SortablePlotItemProps) {
+    const { onPlotUpdate } = props;
+    const secondCurveNeeded = props.plot.type === "differential";
+
+    const handlePlotChange = React.useCallback(
+        function handlePlotChange(changes: Partial<TemplatePlotConfig>) {
+            const newPlot = makeTrackPlot({
+                ...props.plot,
+                ...changes,
+            });
+
+            onPlotUpdate(newPlot);
+        },
+        [props.plot, onPlotUpdate]
+    );
```

Comment:
I dont mind changing it back, but I'd argue that it's fine to just wrap callbacks by default, even if it doesn't do anything. React docs say that there's [no significant harm in doing so](https://react.dev/reference/react/useCallback#should-you-add-usecallback-everywhere), so I'd rather just wrap it in a `callback` as I'm writing, instead of needing to stop and think about whether they're _actually_ gonna be necessary in the end. 


Additionally, it might need to be wrapped in some future update (adding an effect, extracting the menu elements to a sub-component, etc...), so keeping callbacks wrapped can be convenient down the line.

Reply:
I beg to differ. In my opinion, there are some good reasons for not prematurely wrapping all functions in a `useCallback`.

- Unnecessary premature optimization/memory usage (latter admittedly having a rather small impact I assume)
- Depending on how the function is consumed down the line it might actually make the application less performant (might cost more CPU power than not having it, the function is created nonetheless, but you add an additional check) - it only leads to an additional re-render of the descendants if any of them has it somewhere in a dependency array (the descendants re-render anyways if the parent holding the callback re-renders, you can never stop that from happening - apart from memorizing the whole component of course)
- Unnecessary code and dependency arrays for each wrapped function that must be added and require maintenance - true that code might get refactored and true that these functions might get consumed in `useEffect`s but how often does that happen and is it too much to ask of a software engineer to double-check what he puts in a dependency array and/or sends down the line? (we actually also had some support for that in the `ESLint Exhaustive Deps rule` but I don't know if it's still there: https://dev.to/amitaldo/function-makes-the-dependencies-of-useeffect-hook-change-on-every-render-warning-in-react-35hd)
- Worsening readability as all functions get wrapped no matter if they eventually might end up in a dependency array or not (`wsc` is a good example for a codebase where almost everything is (used to be?) wrapped in `useCallback` and `useMemo`, no matter what - I personally find the code very hard to read)
- To break a fly on the wheel just in order to dodge comprehending code/possibly avoid future performance issues does not seem a very good argument to me
- I think you are actually presenting a good case against your statement with the code here as wrapping the functions does not add any benefit at all, just adds more code, and is not even consumed in any sub component (as they are not directly propagated). To argue that everything that is consumed in the HTML part should be wrapped does not make sense to me (or maybe I misunderstand what you're implying) as the HTML parts are not the problem, dependency arrays causing re-renders are. Hence, extracting the elements to a sub-component is per-se not a reason to wrap the functions - using them in a `useEffect` might be.

On the other hand, I admit that it's sometimes hard to know how a function gets consumed down the line, especially in components coming from other packages. In this latter case, I do fully agree that a `useCallback` can come handy. I am not against using `useCallback` per se, just against establishing a rule for wrapping everything in them. I definitely vote for using it with care since I value readability, cleanliness, and maintainability more than premature prevention. 

-IMO